src/LaunchDarkly.ServerSdk/InMemoryFeatureStore.cs

```diff
@@ -15,10 +15,10 @@ namespace LaunchDarkly.Client
     public class InMemoryFeatureStore : IFeatureStore
     {
         private static readonly ILog Log = LogManager.GetLogger(typeof(InMemoryFeatureStore));
-        private readonly ReaderWriterLockSlim RwLock = new ReaderWriterLockSlim();
-        private readonly IDictionary<IVersionedDataKind, IDictionary<string, IVersionedData>> Items =
-            new Dictionary<IVersionedDataKind, IDictionary<string, IVersionedData>>();
-        private bool _initialized = false;
+        private readonly object WriterLock = new object();
+        private volatile ImmutableDictionary<IVersionedDataKind, ImmutableDictionary<string, IVersionedData>> Items =
+            ImmutableDictionary<IVersionedDataKind, ImmutableDictionary<string, IVersionedData>>.Empty;
+        private volatile bool _initialized = false;

```

Comment:
To ensure that other threads see the latest pointer & boolean value we use volatile.